rust/ql/lib/codeql/rust/elements/internal/VariableImpl.qll

```diff
@@ -74,58 +74,66 @@ module Impl {
    * where `definingNode` is the entire `Either::Left(x) | Either::Right(x)`
    * pattern.
    */
-  private predicate variableDecl(AstNode definingNode, AstNode p, string name) {
-    p =
-      any(SelfParam sp |
-        definingNode = sp.getName() and
-        name = sp.getName().getText() and
+  private predicate variableDecl(AstNode definingNode, Name name, string text) {
```

Comment:
This is an orthogonal refactor, but since I was making changes around `Name` it occurred to me that it might be nice to have `Name` here, since both `SelfParam` and `IdentPat` have that in common.